test/widgets/message_list_test.dart

```diff
@@ -1009,4 +1009,61 @@ void main() {
       });
     });
   });
+
+  group('TypingStatus', () {
+    final users = [eg.selfUser, eg.otherUser, eg.thirdUser, eg.fourthUser];
+    final finder = find.descendant(
+      of: find.byType(TypingStatusWidget),
+      matching: find.byType(Text)
+    );
+
+    checkTyping(WidgetTester tester, TypingEvent event, {required String expected}) async {
+      await store.handleEvent(event);
+      await tester.pump();
+      check(finder.evaluate()).single.has((x) => x.widget, 'widget').isA<Text>()
+        .data.equals(expected);
+    }
+
+    testTyping(WidgetTester tester, SendableNarrow narrow) async {
```

Comment:
I think we only normally use the name `testFoo` when the function itself calls `test`/`testWidgets`. I see that `checkTyping` is already taken, though, by the helper just above this. Hmm.

Could find a different name, or maybe not use a helper and just cover the dm and topic cases with something like this (untested):

```dart
for (final (description, narrow, message) in [
  ('dm', …,  …),
  ('topic', …,  …),
]) {
  testWidgets(description, …
```